### DISCUSSION TITLE
Feature Extraction: probabilities

### DISCUSSION THREAD

**Discussion by wmotte** [asreview/asreview#1180] [Q&A] [answered]
Thanks for making this wonderful platform open source!

The export function returns the labels (1, NA, 0) but not the probabilities of the latest Oracle iteration.
For a project identifying rodent-only abstracts, I would like to evaluate the probabilities for all abstracts in the dataset.
Is that possible within the current version?

I know how to load the TF-IDF feature matrix X (from an exported project), but I do not know how to load the latest version of the prediction model (e.g. NB) and apply it to the feature matrix. 
A code snippet would be of great help.

**Comment by PeterLombaers** [comment, marked as answer]
Hey!

ASReview does not store the prediction model or scores, because this would result in a big file size. Instead only the ranking produced by the last model iteration is stored. Not all hope is lost however: if you have the feature matrix and the label data, you can reconstruct the model and use it to calculate the scores again. Something like this works for me:

```[python]
from asreview.config import LABEL_NA

# Choose the classifier and balance strategy that applies to your case.
from asreview.models.classifiers import NaiveBayesClassifier
from asreview.models.balance import DoubleBalance

import numpy as np


classifier = NaiveBayesClassifier()
balance = DoubleBalance()

# Fake feature matrix and label data for illustrative purpose:
feature_matrix = np.random.random((200, 10))
labeled_records = [10, 134, 31, 197, 3]
labels = [0, 1, 1, 0, 1]


y = [LABEL_NA for _ in range(len(feature_matrix))]
for i in range(len(labeled_records)):
    y[labeled_records[i]] = labels[i]

X_train, y_train = balance.sample(feature_matrix, np.array(y), np.array(labeled_records))

classifier.fit(X_train, y_train)
prediction_scores = classifier.predict_proba(feature_matrix)[:, 1]
```

For more information on how the classifier and balance strategy can be used, see the [API reference](https://asreview.readthedocs.io/en/latest/reference.html), or look directly at the pages [here](https://github.com/asreview/asreview/tree/master/asreview/models) on Github.

**Reply by wmotte** [reply]
Many thanks, works for me as well!
